### PR TITLE
[WIP] Fix Bug 1483255 - move search shortcut pinning/unpinning to feed

### PR DESCRIPTION
**Pull request by rlr**
@Mardak @AdamHillier @k88hudson This seems to work (for me?!). I am adding back the telemetry pings and writing some tests... But f?

### CODE REVIEW

File: lib/TopSitesFeed.jsm [hunk 1]
```diff
@@ -612,6 +612,48 @@ this.TopSitesFeed = class TopSitesFeed {
...
   }
 
+  updatePinnedSearchShortcuts({selectedShortcuts}) {
+    const {rows} = this.store.getState().TopSites;
+
+    // Unpin any shortcuts that are no longer selected.
+    rows.filter(row => row && row.isPinned && row.searchTopSite).forEach(({label, url, hostname}) => {
+      if (!selectedShortcuts.find(shortcut => shortcut.label === label)) {
+        NewTabUtils.pinnedLinks.unpin({url});
+        this.store.dispatch(ac.UserEvent({
```
Comment (k88hudson) [inline lib/TopSitesFeed.jsm]: Obviously we'd want to keep this in content

File: lib/TopSitesFeed.jsm [hunk 2]
```diff
@@ -612,6 +612,48 @@ this.TopSitesFeed = class TopSitesFeed {
...
+          // pinnedLinks can have less slots than the total available.
+          index = NewTabUtils.pinnedLinks.links.length;
+        }
+        if (index >= 0) {
+          NewTabUtils.pinnedLinks.pin(shortcut, index);
+        } else {
+          // No slots available, we need to do an insert in first slot and push over other pinned links.
+          this._insertPin(shortcut, 0, numberOfSlots);
+        }
+        this.store.dispatch(ac.UserEvent({
```
Comment (k88hudson) [inline lib/TopSitesFeed.jsm]: Same here

**Comment by k88hudson** [comment]
@rlr this looks fine / works with manual testing, however I wouldn't move the telemetry into the main process as we want user events to come from content (we want the session to be associated, for example)

File: lib/TopSitesFeed.jsm (hunk 1 above)
Comment (rlr) [reply lib/TopSitesFeed.jsm]: oh it needs to be from content? hmm then we'll have to have a bunch of duplicate logic about what we're pinning and unpinning... ? gah

**Comment by rlr** [comment]
closing in favor of #4347

**Pull request closed without merging**